pkg/proto/datadog/trace/idx/span.proto

```diff
@@ -0,0 +1,153 @@
+syntax = "proto3";
+
+package datadog.trace.idx;
+
+option go_package="pkg/proto/pbgo/trace/idx";
+
+message AnyValue {
+	oneof value {
+		// @gotags: json:"string_value" msg:"string_value"
+		uint32 stringValue = 1;
+		// @gotags: json:"bool_value" msg:"bool_value"
+		bool boolValue = 2;
+		// @gotags: json:"double_value" msg:"double_value"
+		double doubleValue = 3;
+		// @gotags: json:"int_value" msg:"int_value"
+		int64 intValue = 4;
+		// @gotags: json:"bytes_value" msg:"bytes_value"
+		bytes bytesValue = 5;
+		// @gotags: json:"array_value" msg:"array_value"
+		ArrayValue arrayValue = 6;
+		// @gotags: json:"key_value_list" msg:"key_value_list"
+		KeyValueList keyValueList = 7;
+	}
+}
+
+message KeyValue {
+	// @gotags: json:"key" msg:"key"
+	uint32 key = 1;
+	// @gotags: json:"value" msg:"value"
+	AnyValue value = 2;
+}
+
+// a repeated list of AnyValue that is needed since `oneof` in AnyValue
+// cannot be `repeated`
+message ArrayValue {
+	// @gotags: json:"values" msg:"values"
+	repeated AnyValue values = 1;
+}
+
+// a repeated list of KeyValue messages that is needed since `oneof`
+// in AnyValue cannot be `repeated` or `map`
+message KeyValueList {
+	// @gotags: json:"key_values" msg:"key_values"
+	repeated KeyValue keyValues = 1;
+}
+
+message SpanLink {
+   // @gotags: json:"trace_id" msg:"trace_id"
+   bytes traceID = 1;                         // Required.
+   // @gotags: json:"span_id" msg:"span_id"
+   fixed64 spanID = 2;                          // Required.
+   // @gotags: json:"attributes" msg:"attributes"
+   repeated KeyValue attributes = 3;
+   // @gotags: json:"tracestate" msg:"tracestate"
+   uint32 tracestate = 4;                      // Optional. W3C tracestate.
+   // @gotags: json:"flags" msg:"flags"
+   uint32 flags = 5;                           // Optional. W3C trace flags. If set, the high bit (bit 31) must be set.
+}
+
+message SpanEvent {
+  // @gotags: json:"time" msg:"time"
+  fixed64 time = 1; // time is the number of nanoseconds between the Unix epoch and this event.
+  // @gotags: json:"name" msg:"name"
+  uint32 name = 2; // name is this event's name.
```

Comment:
Similar to the other comment, I think all fields that are string references could be more descriptive to avoid confusion. It should be very clear when a `uint32` is a string key, and when an actual integer. Something like `nameKey`, `nameRef`, or `nameIdx` maybe?

(This suffix suggestion is not for this field only, just a general observation applicable to all related fields)

Reply:
Yeah definitely, this also frees up those names for helper methods like `func Name() string` for ease of use